dashboard/app/models/foorm/form.rb

```diff
@@ -43,13 +43,32 @@ def self.setup
 
     transaction do
       reset_db
-      Foorm::Form.import! foorms
+      Foorm::Form.import! forms
     end
   end
 
-  def self.get_form_and_latest_version_for_name(form_name)
+  def self.get_questions_and_latest_version_for_name(form_name)
     latest_version = Foorm::Form.where(name: form_name).maximum(:version)
     form = Foorm::Form.where(name: form_name, version: latest_version).first
-    return form, latest_version
+
+    # Substitute any questions from the library.
+    questions = JSON.parse(form.questions)
+    questions["pages"].each do |page|
+      page["elements"].map! do |element|
+        if element["type"] == "library_item"
+          JSON.parse(
+            Foorm::LibraryQuestion.where(
+              library_name: element["library_name"],
+              library_version: element["library_version"].to_i,
+              question_name: element["name"]
+            ).first.question
+          )
+        else
+          element
+        end
+      end
+    end
+
+    return questions, latest_version
   end
```

Comment:
@molly-moen What do you think of this proposal to return the parsed questions as a ruby hash here, rather than returning an entire form object?  If we want to maintain existing behaviour, we can, but we'd need to reencode the modifed questions as JSON again and update the form object before returning it...

Reply:
this makes sense to me since we always parse the questions immediately anyway. If we find a case where we need the raw form we can always add another method that wraps this one.